api/acl.go

```diff
@@ -31,7 +31,7 @@ func (a *ACLPolicies) Upsert(policy *ACLPolicy, q *WriteOptions) (*WriteMeta, er
 	if policy == nil || policy.Name == "" {
 		return nil, errors.New("missing policy name")
 	}
-	wm, err := a.client.write("/v1/acl/policy/"+policy.Name, policy, nil, q)
+	wm, err := a.client.put("/v1/acl/policy/"+policy.Name, policy, nil, q)
```

Comment:
Sorry for sneaking in this little refactor... I don't think I've ever noticed we only ever did PUTs! I really don't want to refight the REST Wars, but I just couldn't bear to put my explicitly not-PUT-like API behind a PUT.